### PR TITLE
Configuration: Get environment from RAILS_ENV, too

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR changes configuration defaults.

The **default value for environment** is also read from `RAILS_ENV`, if `RACK_ENV` can't be found.

## Overview

See #1949

The goal of that Issue is that configuration defaults in [puma/puma-heroku](https://github.com/puma/puma-heroku/blob/master/lib/puma/plugin/heroku.rb) move into [lib/puma/configuration.rb](https://github.com/puma/puma/blob/master/lib/puma/configuration.rb).

### CODE REVIEW

File: test/test_cli.rb [hunk 1]
```diff
@@ -313,11 +313,20 @@ def test_load_path
...
   end
 
-  def test_environment
+  def test_environment_rack_env
     ENV.delete 'RACK_ENV'
 
     Puma::CLI.new ["--environment", @environment]
 
-    assert_equal ENV['RACK_ENV'], @environment
+    assert_equal @environment, ENV['RACK_ENV']
```
Comment (olleolleolle) [inline test/test_cli.rb]: This improves the output on failing  - the order of arguments is: `assert_equal exp, actual`

**Comment by nateberkopec** [comment]
@olleolleolle friendly nudge/reminder to fix the failing tests

**Comment by olleolleolle** [comment]
Huh, it's just like I forgot to push one of the files. I'll get to my computer and investigate!

File: lib/puma/control_cli.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ def initialize(argv, stdout=STDOUT, stderr=STDERR)
       @control_auth_token = nil
       @config_file = nil
       @command = nil
-      @environment = ENV['RACK_ENV']
+      @environment = ENV['RACK_ENV'] || ENV['RAILS_ENV'] || 'development'
```
Comment (olleolleolle) [inline lib/puma/control_cli.rb]: This one! This feels dubious!

**Comment by olleolleolle** [comment]
@nateberkopec There, I think this is ready for a Squash now.

**Pull request merged**